repair/row_level.cc

```diff
@@ -979,6 +998,42 @@ class repair_meta {
         return with_gate(_gate, [this, estimated_partitions] {
             _estimated_partitions = estimated_partitions;
             _repair_writer->set_estimated_partitions(_estimated_partitions);
+            _repair_reader.emplace(_db,
```

Comment:
Creating the reader is not directly related to set_estimated_partitions. I think we have a better place to delay the creation of the reader. We can do that in read_rows_from_disk:
```
read_rows_from_disk() {
       if (!_repair_reader) {
              create the reader
       }
       ....
}
```
This guarantees we always create the reader before we read it.